kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemView.java

```diff
@@ -89,7 +89,7 @@ private FileSystemView(FileSystemView<E> view, Constraints c) {
   public DatasetWriter<E> newWriter() {
     AbstractDatasetWriter<E> writer;
     if (dataset.getDescriptor().isPartitioned()) {
-      writer = new PartitionedDatasetWriter<E>(this);
+      writer = PartitionedDatasetWriter.newWriter(this);
```

Comment:
Which case do you mean? FileSystemWriter.newWriter and PartitionedDatasetWriter.newWriter have the same logic now.